### PR TITLE
M3-4787 Add: modal for upgrading LKE minor version

### PR DESCRIPTION
**Pull request by Jskobos**
## Description

Add modal for bumping LKE version. Only one version bump is supported per PUT (1.16 -> 1.17 but not 1.16 -> 1.18). Copy and some UX details are still under discussion.

## Note to Reviewers

- To test, create an LKE cluster with a version lower than the current (1.18). On the cluster detail page there should be a modal prompting you to upgrade.
- I also updated the mocks so if you turn those on the flow should work.
- Does the transition between 2 modals bother anyone else? Looking for a way around it

Still to come: Banner on LKE landing page and "upgrade" badge in individual cluster rows.

### CODE REVIEW

File: packages/manager/src/features/Kubernetes/KubernetesClusterDetail/UpgradeKubernetesVersionBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+  const { clusterID, currentVersion } = props;
+  const classes = useStyles();
+  const { data: versions } = useKubernetesVersionQuery();
+  const nextVersion = getNextVersion(currentVersion, versions ?? []);
+  const { enqueueSnackbar } = useSnackbar();
+
+  const { updateKubernetesCluster } = useKubernetesClusters();
+
+  const onSubmitUpgradeDialog = (nextVersion: string) => {
+    return updateKubernetesCluster(clusterID, {
```
Comment (johnwcallahan) [inline packages/manager/src/features/Kubernetes/KubernetesClusterDetail/UpgradeKubernetesVersionBanner.tsx:61]: Need to `catch` this (currently POSTing to Sentry).

File: packages/manager/src/features/Kubernetes/KubernetesClusterDetail/UpgradeKubernetesVersionBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+      k8s_version: nextVersion
+    }).then(_ => {
+      recycleNodesDialog.openDialog(undefined);
+    });
+  };
+
+  const onSubmitRecycleDialog = () =>
+    recycleClusterNodes(clusterID).then(_ =>
```
Comment (johnwcallahan) [inline packages/manager/src/features/Kubernetes/KubernetesClusterDetail/UpgradeKubernetesVersionBanner.tsx:69]: Same here, need to `catch` it.

File: packages/manager/src/features/Kubernetes/KubernetesClusterDetail/UpgradeKubernetesVersionBanner.tsx [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+}));
+
+interface Props {
+  clusterID: number;
+  currentVersion: string;
+}
+
+export type CombinedProps = Props;
+
+export const getNextVersion = (
```
Comment (johnwcallahan) [inline packages/manager/src/features/Kubernetes/KubernetesClusterDetail/UpgradeKubernetesVersionBanner.tsx:37]: Would be great to write some unit tests for this function after other functionality/copy is taken care of.

**Review by johnwcallahan — APPROVED** [review]
Great job putting this together so quickly. Left a couple of minor comments.

> Does the transition between 2 modals bother anyone else? Looking for a way around it

It's not horrible, but yeah it does bother me a bit. I wonder if opening the second modal on a slight delay would help? @tiffwong please chime in here.

The other transition that slightly bothers me is that if you're upgrading from 1.16 -> 1.17, on the second modal the banner changes in the background to reflect the newer version (1.18). Not sure if there's a good way around this though.

**Review by tiffwong — APPROVED** [review]
Not a fan of the two modals either but otherwise it looks good, LGTM

**Pull request merged**